### PR TITLE
CONSOLE-3153: Expose timestamp component as part of the dynamic plugin sdk API

### PR DESCRIPTION
**Pull request by rawagner**
The PR exposes Timestamp component as a part of the SDK API.
The Timestamp uses redux to sync all Timestamp component instances at the same time so it is a good candidate for the SDK API package

### CODE REVIEW

**Comment by rawagner** [comment]
@batzionb
@yaacov

**Comment by spadgett** [comment]
@sg00dwin FYI since you're currently assigned https://issues.redhat.com/browse/CONSOLE-3153

File: frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts [hunk 1]
```diff
@@ -248,3 +249,16 @@ export const usePrometheusPoll: UsePrometheusPoll = (props) => {
   // unify order with the rest of API
   return [result[0], !result[2], result[1]];
 };
+
+/**
+ * A component to render timestamp.
+ * The timestamps are synchronized between invidual instances of the Timestamp component.
+ * The provided timestamp is formatted according to user locale.
+ *
+ * @prop timestamp - the timestamp to render
```
Comment (spadgett) [inline frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts:258]: We should discuss what formats are accepted

**Review by spadgett — COMMENTED** [review]
The build failed in CI, but I can't find the log :/

File: frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts (hunk 1 above)
Comment (rawagner) [reply frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts]: Ok, I added 
```
The timestamp format should conform to any form accepted by the new Date() constructor
```
Not sure it that is fine, as it describes internal implementation of the Timestamp component, but Im not sure how to better describe the accepted formats.

Comment (spadgett) [reply frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts]: I would say we support ISO 8601 dates as used by Kubernetes or epoch timestamps as a number.

https://tc39.es/ecma262/#sec-date-time-string-format

Comment (rawagner) [reply frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts]: updated

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by yapei** [comment]
Tried to consume `Timestamp` component in console-demo-plugin and verified that the `Timestamp` component is working
```
$ git diff src/components/ListPage.tsx
diff --git a/dynamic-demo-plugin/src/components/ListPage.tsx b/dynamic-demo-plugin/src/components/ListPage.tsx
index 5ff8e9e6f8..9be46afe27 100644
--- a/dynamic-demo-plugin/src/components/ListPage.tsx
+++ b/dynamic-demo-plugin/src/components/ListPage.tsx
@@ -13,6 +13,7 @@ import {
   RowProps,
   ResourceLink,
   TableColumn,
+  Timestamp,
 } from '@openshift-console/dynamic-plugin-sdk';
 
 const columns: TableColumn<K8sResourceCommon>[] = [
@@ -24,6 +25,10 @@ const columns: TableColumn<K8sResourceCommon>[] = [
     title: 'Namespace',
     id: 'namespace',
   },
+  {
+    title: 'Created',
+    id: 'created'
+  }
 ];
 
 const PodRow: React.FC<RowProps<K8sResourceCommon>> = ({ obj, activeColumnIDs }) => {
@@ -35,6 +40,9 @@ const PodRow: React.FC<RowProps<K8sResourceCommon>> = ({ obj, activeColumnIDs })
       <TableData id={columns[1].id} activeColumnIDs={activeColumnIDs}>
         <ResourceLink kind="Namespace" name={obj.metadata.namespace} />
       </TableData>
+      <TableData id={columns[2].id} activeColumnIDs={activeColumnIDs}>
+        <Timestamp timestamp={obj.metadata.creationTimestamp} className="test-timestamp" />
+      </TableData>
     </>
   );
 };
```
On 'List Page', a new column `Created` is rendered correctly

**Review by jtomasek — APPROVED** [review]
/lgtm

**Pull request merged**